adaptive_ai/adaptive_ai/services/model_selector.py

```
@@ -29,12 +29,71 @@ def __init__(
     ) -> None:
         self.prompt_classifier: PromptClassifier = prompt_classifier
-        self._model_capabilities = get_model_capabilities()
-        self._task_mappings = get_task_model_mappings()
+        self._settings = get_settings()
+        self._model_capabilities = self._get_model_capabilities()
+        self._task_mappings = self._get_task_model_mappings()
         # Set max_workers based on CPU count if not specified
         self.max_workers = max_workers or min(32, 4)  # Default to 4 workers
         logger.info(f"ModelSelector initialized with max_workers={self.max_workers}")
 
+    def _get_model_capabilities(self) -> Dict[str, ModelCapability]:
```

Ruff (0.11.9)

Use `dict` instead of `Dict` for type annotation

Replace with `dict`

(UP006)